parser/src/parser/context.rs

```diff
@@ -75,12 +91,15 @@ impl ParserContext {
     }
 
     ///
-    /// Returns a reference to the next token or error if it does not exist.
+    /// Returns a reference to the next SpannedToken or error if it does not exist.
     ///
     pub fn peek(&self) -> Result<&SpannedToken> {
-        self.tokens.last().ok_or_else(|| self.eof())
+        self.curr().ok_or_else(|| self.eof())
     }
 
+    ///
+    /// Returns a reference to the next Token.
+    ///
     pub fn peek_token(&self) -> Cow<'_, Token> {
         self.tokens
```

Comment:
This should call the helper function `curr()` above.
```suggestion
        self.curr()
```